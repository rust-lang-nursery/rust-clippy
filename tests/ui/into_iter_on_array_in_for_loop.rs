// Tracking issue: #3913
#![deny(clippy::into_iter_on_array)]

fn main() {
    for _ in [1, 2, 3].into_iter() {} //~ ERROR equivalent to .iter()
}
