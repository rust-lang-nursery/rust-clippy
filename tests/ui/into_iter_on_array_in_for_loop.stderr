error: this .into_iter() call is equivalent to .iter() and will not move the array
 --> $DIR/into_iter_on_ref.rs:5:24
   |
LL |     for _ in [1, 2, 3].into_iter() {} //~ ERROR equivalent to .iter()
   |                        ^^^^^^^^^ help: call directly: `iter`
   |
   note: lint level defined here
 --> $DIR/into_iter_on_ref.rs:2:9
   |
LL | #![deny(clippy::into_iter_on_array)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

